analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,409 @@
+        if (fTraceCallGraphRegistry.isEmpty()) {
+            InstrumentedCallStackAnalysis callGraphModule;
+            ITmfTrace trace = TmfTraceManager.getInstance().getActiveTrace();
+            Collection<ITmfTrace> traceSet = TmfTraceManager.getTraceSet(trace);
+            for (ITmfTrace traceMember : traceSet) {
+                Iterable<InstrumentedCallStackAnalysis> modules = TmfTraceUtils.getAnalysisModulesOfClass(traceMember, InstrumentedCallStackAnalysis.class);
+                for (InstrumentedCallStackAnalysis module : modules) {
+                    if (module.getId().equals(fcallStackAnalysisMap.get(traceMember.getTraceTypeId()))) {
+                        callGraphModule = module;
+                        callGraphModule.schedule();
+                        fTraceCallGraphRegistry.put(String.valueOf(traceMember.getName()), callGraphModule);
+                        callGraphModule.waitForCompletion();
+                        break;
+                    }
+                }
+            }
+        }
+
+        refreshDiffCG(monitor);
+        return getDifferentialCallGraphProvider();
+
+    }
+
+    private static Collection<ICallStackElement> getLeafElements(CallGraph weightedTree) {
+        Collection<ICallStackElement> elements = weightedTree.getElements();
+        List<ICallStackElement> leafGroups = new ArrayList<>();
+        for (ICallStackElement group : elements) {
+            leafGroups.addAll(getLeafElement(group));
+        }
+        return leafGroups;
+
+    }
+
+    private static List<ICallStackElement> getLeafElement(ICallStackElement group) {
+        if (group.isLeaf()) {
+            return Collections.singletonList(group);
+        }
+        List<ICallStackElement> leafGroups = new ArrayList<>();
+        group.getChildrenElements().forEach(g -> leafGroups.addAll(getLeafElement(g)));
+        return leafGroups;
+    }
+
+    private static void recurseAddElementData(ICallStackElement element, String group, CallGraph callGraph, WeightedTreeSet<ICallStackSymbol, Object> newTreeSet) {
+        // Add the current level of trees to the new tree set
+        for (AggregatedCallSite tree : callGraph.getCallingContextTree(element)) {
+            newTreeSet.addWeightedTree(group, tree.copyOf());
+        }
+
+        // Recursively add the next level of elements
+        ICallStackElement treeEl = element;
+        Collection<ITree> children = treeEl.getChildren();
+        for (ITree child : children) {
+            if (child instanceof ICallStackElement) {
+                recurseAddElementData((ICallStackElement) child, group, callGraph, newTreeSet);
+            }
+        }
+    }
+
+    @Override
+    protected boolean executeAnalysis(IProgressMonitor monitor) throws TmfAnalysisException {
+        return true;
+    }
+
+    @Override
+    public boolean waitForCompletion() {
+        return true;
```

Comment:
Should be implemented correctly look at other analyses implementation to see how it is typically done.